dao/mongodb.rya/src/main/java/mvm/rya/mongodb/MongoConnectorFactory.java

```diff
@@ -41,26 +47,77 @@
+            if (conf.getBoolean(MongoDBRdfConfiguration.USE_TEST_MONGO, false)) {
+                try {
+                    MongodForTestsFactory testsFactory = MongodForTestsFactory.with(Version.Main.PRODUCTION);
+                    mongoClient = testsFactory.newMongo();
+                } catch (IOException e) {
+                    // Rethrow as an unchecked error.  Since we are in a test mode here, just fail fast.
+                    throw new MongoException("While creating a factory for a test/mock MongoDB instance.",e);
+                }
             } else {
-                mongoClient = new MongoClient(server);
+                // Connect to a running Mongo server
+                final String host = requireNonNull(conf.get(MongoDBRdfConfiguration.MONGO_INSTANCE), "MongoDB host name is required");
+                final int port = requireNonNullInt(conf.get(MongoDBRdfConfiguration.MONGO_INSTANCE_PORT),
+                        "MongoDB Port number is required.");
+                ServerAddress server;
+                try {
+                    server = new ServerAddress(host, port);
+                } catch (UnknownHostException e) {
+                    throw new MongoException("Connecting MongoDB instance, cannot find host="+host,e);
+                }
+                // check for authentication credentials
+                if (conf.get(MongoDBRdfConfiguration.MONGO_USER) != null) {
+                    final String username = conf.get(MongoDBRdfConfiguration.MONGO_USER);
+                    final String dbName = requireNonNull(conf.get(MongoDBRdfConfiguration.MONGO_DB_NAME),
+                            MongoDBRdfConfiguration.MONGO_DB_NAME + " is null but required configuration if "
+                                    + MongoDBRdfConfiguration.MONGO_USER + " is configured.");
+                    final char[] pswd = requireNonNull(conf.get(MongoDBRdfConfiguration.MONGO_USER_PASSWORD),
+                            MongoDBRdfConfiguration.MONGO_USER_PASSWORD + " is null but required configuration if "
+                                    + MongoDBRdfConfiguration.MONGO_USER + " is configured.").toCharArray();
+                    final MongoCredential cred = MongoCredential.createCredential(username, dbName, pswd);
+                    mongoClient = new MongoClient(server, Arrays.asList(cred));
+                } else {
+                    // No user was configured:
+                    mongoClient = new MongoClient(server);
+                }
             }
         }
         return mongoClient;
     }
+
+    /**
+     * Throw exception for un-configured required values.
+     * 
+     * @param required  String to check
+     * @param message  throw configuration exception with this description
+     * @return unaltered required string
+     * @throws ConfigurationRuntimeException  if required is null
+     */
+    static private String requireNonNull(String required, String message) throws ConfigurationRuntimeException {
```

Comment:
In addition to what I said to Aaron:  I am preserving the difference between a missing configuration and a random NullPointerException and NumberFormatException.  
I'll switch it back if you all consider this unimportant.  The method now throws two unchecked, and IMO not recoverable errors:  ConfigurationRuntimeException for configuration parameters that are missing, and MongoException for issues preventing connection, possibly temporary.  But those issues can be specifically caught, for example: to retry.   A NullPointerException could be thrown for any reason.